File: src/dataset/cpp/triangle_exercism-stungeye/openAI/ChatGPT4_triangle_v1.cpp
#include "triangle.h"

namespace triangle {
  flavor kind(double a, double b, double c) {
    if (a <= 0 || b <= 0 || c <= 0) {
      throw std::domain_error("All triangle sides must have a length greater than zero.");
    }

    if (a + b <= c || a + c <= b || b + c <= a) { 
      throw std::domain_error("Combined length of any two triangle sides must be greater than the length of the third side.");
    }

    if (a == b && b == c) {
      return flavor::equilateral;
    }
    if (a == b || b == c || a == c) {
      return flavor::isosceles;
    }

    return flavor::scalene;
  }
}  // namespace triangle

File: src/dataset/typescript/triangle_exercism-wackerow/openAI/ChatGPT4_triangle_v1.ts
export default class Triangle {
  private _sides: number[];

  constructor(...sides: number[]) {
    this._sides = sides;
  }

  public kind(): string {
    const [a, b, c] = this._sides;

    if (
      this._sides.length !== 3 ||
      a <= 0 || b <= 0 || c <= 0 ||
      a + b <= c || b + c <= a || c + a <= b
    ) {
      throw 'Impossible triangle';
    }

    if (a === b && b === c) return 'equilateral';
    if (a === b || b === c || c === a) return 'isosceles';
    return 'scalene';
  }
}